### ISSUE TITLE
error on make fail

### ISSUE THREAD

**Issue by zxybfq** [McMCCRU/rtl8188gu#66]
/root/rtl8188gu1/os_dep/linux/ioctl_cfg80211.c: In function ‘rtw_cfg80211_ch_switch_notify’:
/root/rtl8188gu1/os_dep/linux/ioctl_cfg80211.c:398:9: error: too few arguments to function ‘cfg80211_ch_switch_notify’
  398 |         cfg80211_ch_switch_notify(adapter->pnetdev, &chdef, 0);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /root/rtl8188gu1/include/osdep_service_linux.h:93,
                 from /root/rtl8188gu1/include/osdep_service.h:47,
                 from /root/rtl8188gu1/include/drv_types.h:27,
                 from /root/rtl8188gu1/os_dep/linux/ioctl_cfg80211.c:17:
/usr/src/linux-headers-6.5.0-kali2-common/include/net/cfg80211.h:8575:6: note: declared here
 8575 | void cfg80211_ch_switch_notify(struct net_device *dev,
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
/root/rtl8188gu1/os_dep/linux/ioctl_cfg80211.c: In function ‘cfg80211_rtw_scan’:
/root/rtl8188gu1/os_dep/linux/ioctl_cfg80211.c:2619:33: warning: the comparison will always evaluate as ‘true’ for the address of ‘ssid’ will never be NULL [-Waddress]
 2619 |                 if (ssids->ssid != NULL
      |                                 ^~
/usr/src/linux-headers-6.5.0-kali2-common/include/net/cfg80211.h:2442:12: note: ‘ssid’ declared here
 2442 |         u8 ssid[IEEE80211_MAX_SSID_LEN];
      |            ^~~~
make[3]: *** [/usr/src/linux-headers-6.5.0-kali2-common/scripts/Makefile.build:248：/root/rtl8188gu1/os_dep/linux/ioctl_cfg80211.o] 错误 1
make[2]: *** [/usr/src/linux-headers-6.5.0-kali2-common/Makefile:2057：/root/rtl8188gu1] 错误 2
make[1]: *** [/usr/src/linux-headers-6.5.0-kali2-common/Makefile:246：__sub-make] 错误 2
make[1]: 离开目录“/usr/src/linux-headers-6.5.0-kali2-amd64”
make: *** [Makefile:1895：modules] 错误 2

**Comment by dubhater** [comment]
Your device should work out of the box starting with kernel 6.4. Does it not work?

**Comment by idonashino** [comment]
my linux kernel is 6.4, and i also meet this problem. Have you solved it?

**Comment by a5a5aa555oo** [comment]
rtl8188gu has two modes: CDROM mode and WiFi mode, type `lsusb` in the terminal to see which mode it is in.
if your adapter is in CDROM mode, install `usb_modeswitch` to switch it to WiFi mode.

**Comment by brucelelele** [comment]
Same issue, have you figure out?

**Comment by brucelelele** [comment]
It may seems like a compiling issue for me. Because when I do the guide I get the same result as you from it using Kernel 6.5. Then I downgrade to 6.3 only seems to get a kernel headers issue.

**Comment by a5a5aa555oo** [comment]
Linux kernel uses `rtl8xxxu` to drive rtl8188gu since 6.4, you guys blacklist it?

**Comment by brucelelele** [comment]
> Linux kernel uses `rtl8xxxu` to drive rtl8188gu since 6.4, you guys blacklist it?

You mean this Repo doesn't need to add to Linux now?  How to know if I had that rtl8xxxu? Cuz obviously I didn't set rtl8188gu successfully on linux6.5

**Comment by a5a5aa555oo** [comment]
Just load rtl8xxxu if you are using linux kernel 6.4 and higher.
`sudo modprobe rtl8xxxu`

if rtl8xxxu is not loaded automatically on boot, check your /etc/modprobe.d/ directory if there is something preventing rtl8xxxu from loading.

Check also if your system has the firmware files needed.
`ls /lib/firmware/rtlwifi/rtl8710bufw*`

if not, download and copy rtl8710bufw_UMC.bin/rtl8710bufw_SMIC.bin to /lib/firmware/rtlwifi/
https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/rtlwifi

**Comment by brucelelele** [comment]
> Just load rtl8xxxu if you are using linux kernel 6.4 and higher.
> `sudo modprobe rtl8xxxu`
> 
> if rtl8xxxu is not loaded automatically on boot, check your /etc/modprobe.d/ directory if there is something preventing rtl8xxxu from loading.

![20231017141559(1).png](https://github.com/McMCCRU/rtl8188gu/assets/111379118/1731f1a7-e4b6-454d-823e-337a8eacd0e0)

![20231017141833(1).png](https://github.com/McMCCRU/rtl8188gu/assets/111379118/54a9f618-aa68-4a37-89c7-3096ef855647)

**Comment by a5a5aa555oo** [comment]
what are the outputs of these commands?

`lsusb`
`sudo dmesg | grep rtl8xxxu`

**Comment by brucelelele** [comment]
> what are the outputs of these commands?
> 
> `lsusb`
> `sudo dmesg | grep rtl8xxxu`

![20231017143024(1).png](https://github.com/McMCCRU/rtl8188gu/assets/111379118/74e2f593-b795-4873-b767-4a57a047371a)

![20231017143121(1).png](https://github.com/McMCCRU/rtl8188gu/assets/111379118/99106150-6e4c-4ca8-98b0-7c0b742f5ef8)

**Comment by brucelelele** [comment]
> what are the outputs of these commands?
> 
> `lsusb`
> `sudo dmesg | grep rtl8xxxu`

![20231017144835.png](https://github.com/McMCCRU/rtl8188gu/assets/111379118/c476d003-a33a-49cf-b544-6a329447c9f5)

![20231017143024(1).png](https://github.com/McMCCRU/rtl8188gu/assets/111379118/7b279350-7fec-4b69-bfa7-71d4055ece07)

**Comment by a5a5aa555oo** [comment]
still not working? run this command and post the link.
`sudo dmesg | nc termbin.com 9999`

**Comment by brucelelele** [comment]
> still no working? run this command and post the link.
> `sudo dmesg | nc termbin.com 9999`

Yeah, tricky![20231017151213(1).png](https://github.com/McMCCRU/rtl8188gu/assets/111379118/b5ed7c77-1c32-4ffe-a6a2-bb0ac15ab0f7)

**Comment by lsclsclsc** [comment]
> It may seems like a compiling issue for me. Because when I do the guide I get the same result as you from it using Kernel 6.5. Then I downgrade to 6.3 only seems to get a kernel headers issue.
thanks my friend

**Comment by a5a5aa555oo** [comment]
unload 8188gu/rtl8xxxu both and try again.
`sudo modprobe -r rtl8xxxu`
`sudo modprobe -r 8188gu`
`sudo modprobe rtl8xxxu`

if it works, remove the 8188gu driver you installed before.
if it doesn't work, post the full log of dmesg again.

**Comment by zxybfq** [comment]
Thank you, you have given me great ideas and solutions

**Comment by brucelelele** [comment]
> unload 8188gu/rtl8xxxu both and try again.
> `sudo modprobe -r rtl8xxxu`
> `sudo modprobe -r 8188gu`
> `sudo modprobe rtl8xxxu`
> 
> if it works, remove the 8188gu driver you installed before.
> if it doesn't work, post the full log of dmesg again.

Thank you very much. It now work. I think your answer is currently the only few workable solution for this chip this kernel version for me at this time .

**Comment by a5a5aa555oo** [comment]
glad it works now. remember to blacklist or remove 8188gu.

`sudo sh -c 'echo blacklist 8188gu > /etc/modprobe.d/blacklist-8188gu.conf'`

**Comment by Panpan021** [comment]
how to solve this problem?
![image](https://github.com/McMCCRU/rtl8188gu/assets/106633832/ca2435ef-50a9-4b17-a68e-d94764afbdbb)

**Comment by dubhater** [comment]
You need to install the package which provides that file. It may be called „firmware-realtek”. Alternatively, you can download the file straight from the source: https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/plain/rtlwifi/rtl8710bufw_UMC.bin

**Comment by Panpan021** [comment]
> You need to install the package which provides that file. It may be called „firmware-realtek”. Alternatively, you can download the file straight from the source: https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/plain/rtlwifi/rtl8710bufw_UMC.bin

Thank you! It works!
